### ISSUE TITLE
Can't start app using pdfrx ^0.4.10

### ISSUE THREAD

**Issue by mgeilich** [espresso3389/pdfrx#88]
[CXX1300] CMake '3.18.1' was not found in SDK, PATH, or by cmake.dir property.
[CXX1416] Could not find Ninja on PATH or in SDK CMake bin folders.

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':pdfrx'.
> java.lang.NullPointerException (no error message)

**Comment by espresso3389** [comment]
You can find identical issue by searching [CMake '3.18.1' was not found in SDK](https://github.com/espresso3389/pdfrx/issues?q=is%3Aissue+CMake+%273.18.1%27+was+not+found+in+SDK) on the issues.

Anyway, short answer is;

https://github.com/espresso3389/pdfrx/issues/48#issuecomment-1926951449

Update your `android/gradle/wrapper/gradle-wrapper.properties` to use newer gradle:

```gradle
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.1-bin.zip
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
```

**Comment by mgeilich** [comment]
I tried the fix above - changing my distributionUrl from:
distributionUrl=https\://services.gradle.org/distributions/gradle-7.5-all.zip

to:
distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.1-bin.zip

but I still get the identical build error:

Building with Flutter multidex support enabled.
[CXX1300] CMake '3.18.1' was not found in SDK, PATH, or by cmake.dir property.
[CXX1416] Could not find Ninja on PATH or in SDK CMake bin folders.

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':pdfrx'.
> java.lang.NullPointerException (no error message)

**Comment by espresso3389** [comment]
It seems [connectivity_plus 5.0.2](https://pub.dev/packages/connectivity_plus/versions/5.0.2) does not support Flutter 3.19 so far. Please use [pdfrx 0.4.46+](https://pub.dev/packages/pdfrx/versions/0.4.46) as [noted on the README.md](https://github.com/espresso3389/pdfrx?tab=readme-ov-file#pdfrx);

```yaml
  pdfrx: ^0.4.46
```

**Comment by espresso3389** [comment]
And, for the original one, CMake issue, did you try `flutter clean` before rebuilding?

**Comment by espresso3389** [comment]
If the problem persists, please show me your `flutter doctor -v` anyway.

**Comment by mgeilich** [comment]
Thanks for the support!

Yes I did use flutter clean; flutter pub get before rebuilding with 0.4.10.

I get the same package conflict with 0.4.46 as I do with 0.4.11:
Because no versions of connectivity_plus match >5.0.2 <6.0.0 and connectivity_plus 5.0.2 depends on js
  ^0.6.4, connectivity_plus ^5.0.2 requires js ^0.6.4.
And because pdfrx >=0.4.11 <1.0.7 depends on js ^0.7.0, connectivity_plus ^5.0.2 is incompatible with pdfrx
  >=0.4.11 <1.0.7.
So, because occucore depends on both connectivity_plus ^5.0.2 and pdfrx ^0.4.46, version solving failed.

My flutter doctor -v says:

[✓] Flutter (Channel stable, 3.19.1, on macOS 14.1.1 23B81 darwin-arm64, locale en-US)
    • Flutter version 3.19.1 on channel stable at /Users/michael/Desktop/Edare/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision abb292a07e (9 days ago), 2024-02-20 14:35:05 -0800
    • Engine revision 04817c99c9
    • Dart version 3.3.0
    • DevTools version 2.31.1

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0-rc1)
    • Android SDK at /Users/michael/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C500b
    • CocoaPods version 1.14.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.87.0)
    • VS Code at /Users/michael/Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.82.0

[✓] Connected device (3 available)
    • iPad (2) (mobile) • 00008103-001471841120801E • ios            • iOS 17.3.1 21D61
    • macOS (desktop)   • macos                     • darwin-arm64   • macOS 14.1.1 23B81 darwin-arm64
    • Chrome (web)      • chrome                    • web-javascript • Google Chrome 120.0.6099.234
    ! Error: Browsing on the local area network for iPhone (38). Ensure the device is unlocked and attached
      with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)

[✓] Network resources
    • All expected network resources are available.

**Comment by espresso3389** [comment]
I've released [0.4.48](https://pub.dev/packages/pdfrx/versions/0.4.48) that rollbacks js to 0.6.7.
Please check it.

**Comment by mgeilich** [comment]
That worked!
